Adds support for Jobs

zhill opened the pull request:
Adds Job support. Resolves #30

luhring commented:
Closes #30

luhring commented on cmd/kubernetes-admission-controller/admission/hook_test.go:
```diff
@@ -101,6 +102,7 @@ func TestHook_Validate(t *testing.T) {
 			admissionRequest:      podAdmissionRequest(t, nonexistentImageName),
 			isExpectedToBeAllowed: true,
 		},
+		// Deployment resource tess
```
nit: sp. `tess` -> `tests`

luhring commented on cmd/kubernetes-admission-controller/extractor/from_job_test.go:
```diff
@@ -0,0 +1,53 @@
+		Annotations: map[string]string{
+			"annotation1": "value1",
+		},
+		Name:      "a_deployment",
```
This shouldn't affect the tests, but my original idea was to have the names match the kind of object being defined, e.g. `a_job`. But don't worry about this, I have some other ideas to mitigate the copy/paste required for new tests.

luhring reviewed (approved):
Very nice work! 🎉  And it's very informative to see the pattern playing out with new kinds of resources. I've noticed a few opportunities for an improvement to what I had put in place in the last PR, to help reduce copy/paste needed.

zhill commented:
I addressed feedback but updated the branch name to issue-30, so I'll close this and open a new PR from that branch where I also added the issue reference in the commit.
